pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -28,6 +28,21 @@ const (
 	PrometheusKindKey = "prometheus"
 )
 
+// PrometheusInterface is used by Prometheus and PrometheusAgent to share common methods, e.g. config generation.
+// +k8s:deepcopy-gen=false
+type PrometheusInterface interface {
+	GetCommonPrometheusFields() CommonPrometheusFields
+	GetObjectMeta() metav1.ObjectMeta
```

Comment:
suggestion: leverage the existing [ObjectMetaAccessor](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMetaAccessor) interface.

```suggestion
	metav1.ObjectMetaAccessor
```

Reply:
Oh nice, I wasn't aware this existed :)